src/main/java/org/privacyidea/JSONParser.java

```diff
@@ -184,7 +185,21 @@ public PIResponse parsePIResponse(String serverResponse)
         if (detailElem != null && !detailElem.isJsonNull())
         {
             JsonObject detail = obj.getAsJsonObject(DETAIL);
-            response.preferredClientMode = getString(detail, PREFERRED_CLIENT_MODE);
+
+            // Translate some preferred client mode names
+            String modeFromResponse = getString(detail, PREFERRED_CLIENT_MODE);
+            if (Objects.equals(modeFromResponse, "poll"))
```

Comment:
I don't think it is necessary to use Object.equals, it makes the code more verbose.
`"poll".equals(modeFromResponse)` is simpler. Note that this is null-safe because `"poll"` will always be non-null and `equals` is defined for null inputs.

Reply:
Fixed [71f1a8f](https://github.com/privacyidea/java-client/pull/50/commits/71f1a8f84bf3a15fd9658d38927eb826f303aa17)